### ISSUE TITLE
Fatal error on author archives

### ISSUE THREAD

**Issue by swissspidy** [johnbillion/user-switching#39]
Since 3bb546800938e8ba4ef0bc1f97b50c108b9289bf the plugin uses `current_user_can( 'switch_to_user', get_queried_object_id() ) ` and `self::switch_to_url( get_queried_object() ) )` in the admin bar on author archives.

For some reason, `queried_object` is `false` in my environment when viewing an author archive and `queried_object_id` is `0`.

This leads to a fatal error because of the strict type checking:

```
Uncaught TypeError: Argument 1 passed to user_switching::switch_to_url() must be an instance of WP_User, boolean given, called in /wp-content/plugins/user-switching/user-switching.php on line 487
```

While I am trying to figure out why this is the case (and why `current_user_can( 'switch_to_user', 0 )` works), I thought I'd raise this here just in case.

It can't hurt to have some type checking before calling that function.

**Comment by johnbillion** [comment]
Thanks for the report. I'm really interested in the root cause of that empty queried object on an author archive.

**Comment by swissspidy** [comment]
Weird. I somehow traced this down to Jetpack's Infinite Scroll module.

Calling `get_queried_object()` when filtering `infinite_scroll_archive_supported` causes this error. An example:

```php
function infinite_scroll_support( $supported ) {
	if ( get_queried_object() ) {
		return false;
	}

	return $supported;
}

add_filter( 'infinite_scroll_archive_supported', 'infinite_scroll_support' );
```

**Comment by johnbillion** [comment]
Does Jetpack mess with the `WP_Query->is_*` properties anywhere? That could be a cause.

I don't enjoy fatals but I also don't like papering over lower level problems. User Switching should be able to trust that `get_queried_object()` is a `WP_User` object when `is_author()` is true. Ref: https://github.com/WordPress/wordpress-develop/blob/0fc5c3adaf99ad4f81b10dbd93adc83141231b4f/src/wp-includes/class-wp-query.php#L3477-L3480

**Comment by johnbillion** [comment]
Wanna report this upstream to Jetpack?

**Comment by johnbillion** [comment]
Fixed in 1.5.1.

**Comment by swissspidy** [comment]
Thanks a lot John! And sorry for not following up with your suggestion about reporting this upstream.
